istioctl/cmd/analyze.go

```diff
@@ -172,13 +194,30 @@ istioctl experimental analyze -k -d false
 				}
 			}
 
-			// Print validation message output, or a line indicating that none were found
-			if len(outputMessages) == 0 {
-				fmt.Fprintln(cmd.ErrOrStderr(), "\u2714 No validation issues found.")
-			} else {
-				for _, m := range outputMessages {
-					fmt.Fprintln(cmd.OutOrStdout(), renderMessage(m))
+			switch msgOutputFormat {
+			case LogOutput:
+				// Print validation message output, or a line indicating that none were found
+				if len(outputMessages) == 0 {
+					fmt.Fprintln(cmd.ErrOrStderr(), "\u2714 No validation issues found.")
+				} else {
+					for _, m := range outputMessages {
+						fmt.Fprintln(cmd.OutOrStdout(), renderMessage(m))
+					}
+				}
+			case JsonOutput:
+				jsonOutput, err := json.Marshal(outputMessages)
```

Comment:
should use MarshalIndent to make the output readable

Reply:
I'd rather not by default--it's a lot easier to add indentation with e.g. `| jq` than it is to get rid of it. I'm not philosophically against adding it later if a user comes up with a reasonable issue though.

Reply:
I say marshal indent because it is how istioctl commands already work, how kubectl works, and majority of the time this is run in a CLI case not an API where we need to save bytes

Reply:
+1 to nicely formatted output as the default. Compact output can still be achieved with `jq . --compact-output`.